python/dgl/convert.py

```diff
@@ -246,6 +246,51 @@ def hetero_from_relations(rel_graphs):
     -------
     DGLHeteroGraph
         A heterograph consisting of all relations.
+
+    Notes
+    -----
```

Comment:
A note is usually short side information. This one is a pretty long note. Overall, I feel it might be more helpful to give a more informative error message. Example is `Error: graph 'user-follows-user' and graph 'user-plays-game' should have the same number of 'user' nodes. Got XX and YY.`.

Also, the word "workaround" is for something that is wrong not for an intended behavior.

Reply:
I can make this an example rather than a note.

I understand this is the intended behavior for current system design, but it's also quite natural to expect the system to automatically take the maximum number of nodes from modeling perspective. As a user who does modeling, I'd expect common questions/issues addressed in the documentation.

Reply:
The thing is that the note is about how to create a heterograph by providing an explicit cardinality. It is not about the `hetero_from_relations` API. A better place should be in the `dgl.bipartite` or `dgl.graph` API.

How about putting the whole section in the beginning since it is quite important. Consider the following:
```
'''Create a heterograph from graphs representing connections of each relation.

The input is a list of heterographs where the i-th graph contains edges of type
`(s_i, e_i, d_i)`. For any two graphs i and j, if s_i is equal to s_j,
we check that `rel_graphs[i].number_of_nodes(s_i) == rel_graphs[j].number_of_nodes(s_j)`.
Namely, if two graphs share the same node type, the number of nodes of that type
should match too. This also applies to destination node type.

Parameters
------------
...

Examples
----------
(put your example here to show that explicit cardinality is sometimes necessary)
'''
```

Reply:
Done.